ViAn/GUI/Analysis/analysisslider.cpp

```diff
@@ -52,20 +52,39 @@ void AnalysisSlider::paintEvent(QPaintEvent *ev) {
         }
     }
 
+    // Draws the analysis interval
+    if (show_ana_interval) {
+        brush = Qt::darkMagenta;
+        double first = (groove_rect.left()+(double)m_ana_interval.first*c);
+        double second = (groove_rect.left()+(double)m_ana_interval.second*c);
+        if (m_ana_interval.first != -1 && m_ana_interval.second != -1 && m_ana_interval.first <= m_ana_interval.second) {
```

Comment:
It quite hard to read the code with all these conditions that doesn't really say anything. 

 m_ana_interval.first != -1 && m_ana_interval.second != -1

Could for example be refactored into a "valid_interval" function or similar.